x-pack/legacy/plugins/lens/public/types.ts

```diff
@@ -330,17 +402,17 @@ export interface Visualization<T = unknown, P = unknown> {
   visualizationTypes: VisualizationType[];
 
   getLayerIds: (state: T) => string[];
-
   clearLayer: (state: T, layerId: string) => T;
-
   removeLayer?: (state: T, layerId: string) => T;
-
   appendLayer?: (state: T, layerId: string) => T;
 
+  // Layer context menu is used by visualizations for styling the entire layer
+  // For example, the XY visualization uses this to have multiple chart types
   getLayerContextMenuIcon?: (opts: { state: T; layerId: string }) => IconType | undefined;
-
   renderLayerContextMenu?: (domElement: Element, props: VisualizationLayerConfigProps<T>) => void;
 
+  getLayerOptions: (props: VisualizationLayerConfigProps<T>) => VisualizationLayerConfigResult;
```

Comment:
For reviewers: This line of code is one of the key changes. Visualizations are no longer allowed to have a `renderDimensionPanel` method, and instead provide an object that defines what the right side panel does https://github.com/elastic/kibana/pull/58279/files#diff-6ae9ccca58dffda903f0e2042bcf455bR258